rocAL/include/readers/webdataset_source_reader.h

```diff
@@ -0,0 +1,145 @@
+    //! Opens the next file in the folder
+    /*!
+     \return The size of the next file, 0 if couldn't access it
+    */
+    size_t open() override;
+    //! Resets the object's state to read from the first file in the folder
+    void reset() override;
+
+    //! Returns the id of the latest file opened
+    std::string id() override { return _last_id; };
+
+    unsigned count_items() override;
+
+    ~WebDatasetSourceReader() override;
+
+    int close() override;
+
+    WebDatasetSourceReader();
+
+    //! Returns the number of images in the last batch
+    size_t last_batch_padded_size() override;
+
+  private:
+    //! opens the folder containnig the images
+    Reader::Status webdataset_record_reader();
+    Reader::Status folder_reading();
+    std::string _folder_path;
+    std::string _path;
+    std::string _index_paths;
+    DIR *_src_dir;
+    DIR *_sub_dir;
+    struct dirent *_entity;
+    std::vector<std::string> _file_names, _all_shard_file_names_padded;
+    std::map<std::string, unsigned int> _file_wds_shard_idx_mapping, _all_shard_wds_shard_idx_mapping_padded;
+    std::map<std::string, unsigned int> _file_size, _file_offset, _all_shard_file_sizes_padded, _all_shard_file_offset_padded;
+    unsigned _curr_file_idx;
+    unsigned _current_file_size;
+    std::string _last_id;
+    std::vector<std::string> _index_name_list;
+    int _file_name_count = 0;
+    //!< _batch_count Defines the quantum count of the images to be read. It's
+    //!< usually equal to the user's batch size.
+    /// The loader will repeat images if necessary to be able to have images
+    /// available in multiples of the load_batch_count, for instance if there
+    /// are 10 images in the dataset and _batch_count is 3, the loader repeats 2
+    /// images as if there are 12 images available.
+    size_t _batch_count = 1;
+    size_t _file_id = 0;
+    size_t _in_batch_read_count = 0;
+    bool _loop;
+    bool _shuffle;
+    int _read_counter = 0;
+    void incremenet_read_ptr();
+    int release();
+    size_t get_file_shard_id();
+    void incremenet_file_id() { _file_id++; }
+    void parse_tar_files(std::vector<SampleDescription> &samples_container, std::vector<ComponentDescription> &components_container, std::unique_ptr<FileIOStream> &tar_file);
+    void parse_index_files(std::vector<SampleDescription> &samples_container,
+                      std::vector<ComponentDescription> &components_container,
+                      const std::string &index_path);
+    void parse_sample_description(
+        std::vector<SampleDescription> &samples_container,
+        std::vector<ComponentDescription> &components_container,
+        std::ifstream &index_file, const std::string &index_path, int64_t line,
+        int index_version);
+    Reader::Status webdataset_record_reader_from_components(ComponentDescription component, unsigned wds_shard_index);
+    std::shared_ptr<MetaDataReader> _meta_data_reader = nullptr;
```

Comment:
Don't add variables in between, Please keep all variables in the beginning following by member functions